core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -586,6 +587,29 @@ private[spark] class MemoryStore(
     }
   }
 
+  /**
+   * Only release memory used by this task from unroll memory
+   * map(`onHeapUnrollMemoryMap` or `offHeapUnrollMemoryMap`).
+   */
+  def releaseMemoryFromUnrollMemoryMap(
+      memoryMode: MemoryMode,
+      memory: Long = Long.MaxValue): Unit = {
+    val taskAttemptId = currentTaskAttemptId()
+    memoryManager.synchronized {
+      val unrollMemoryMap = memoryMode match {
+        case MemoryMode.ON_HEAP => onHeapUnrollMemoryMap
+        case MemoryMode.OFF_HEAP => offHeapUnrollMemoryMap
+      }
+      if (unrollMemoryMap.contains(taskAttemptId)) {
```

Comment:
Possible micro-optimisation: you can use `getOrElse(taskAttemptId, -1)` instead of `contains` then you can decide whether the key is contained and get the value for the key with one lookup.

Reply:
Yeah, it looks better,  thanks